lms/static/js/ccx/view/schedule_date_button.js

```diff
@@ -0,0 +1,129 @@
+;(function (define) {
+    'use strict';
+    define([
+            'backbone',
+            'jquery',
+            'underscore',
+            'gettext',
+            'text!templates/ccx/underscore/date-button.underscore',
+            'edx-ui-toolkit/js/utils/html-utils'
+        ],
+        function (Backbone,
+                  $,
+                  _,
+                  gettext,
+                  scheduleTreeDateButtonTemplate,
+                  HtmlUtils) {
+            return Backbone.View.extend({
+
+                initialize: function (options) {
+                    this.dateType = options.dateType;
+                    this.date = options.date;
+                    this.template = HtmlUtils.template(scheduleTreeDateButtonTemplate);
+                },
+
+                render: function () {
+                    HtmlUtils.setHtml(
+                        this.$el,
+                        this.template({date: this.date})
+                    );
+                    this.location = this.$el.data("location");
+                    var $dateChangeButton = $(this.$el.find("button"));
+
+                    $dateChangeButton.attr('href', '#enter-date-modal').leanModal({closeButton: '.close-modal'});
+                    $dateChangeButton.click(this.editDateCCXSchedule());
+
+                    return this;
+                },
+
+                editDateCCXSchedule: function () {
+                    var self = this;
+                    return function () {
+                        var dateParts;
+                        var row = $(this).closest('tr');
+                        var modal = $('#enter-date-modal')
+                            .data('what', self.dateType)
+                            .data('location', row.data('location'));
+
+                        modal.find('h2').text(
+                            self.dateType === 'due' ? HtmlUtils.ensureHtml(gettext("Enter Due Date and Time")) :
```

Comment:
As mentioned previously, the HtmlUtils.ensureHtml call here is incorrect. If the the translation of this string happens to have HTML in it, we do want it to be escaped.

The JQuery text method escapes its argument (http://api.jquery.com/text/), so you just need

modal.find('h2').text(self.dateType === 'due' ? gettext("Enter Due Date and Time") : gettext("Enter Start Date and Time"));

 I won't comment on further ensureHtml issues in this review pass, but I wanted you to be aware that escaping is handled by the JQuery text method.